src/shared/gitlab-client.ts

```diff
@@ -149,4 +159,77 @@ export class GitlabClient {
     };
   }
 
+  public async getUserByEmail(email: string) {
+    if (!validateEmail(email)) {
+      throw Boom.badRequest(`Invalid email '${email}'`);
+    }
+    const search = {
+      search: email,
+    };
+    const users = await this.fetchJson<User[]>(`users?${qs.stringify(search)}`, true);
+    if (!users || !users.length) {
+      throw Boom.badRequest(`Can\'t find user '${email}'`);
+    }
+    if (users.length > 1) {
+      // This shoud never happen
+      throw Boom.badRequest(`Found multiple users with email '${email}'`);
+    }
+    return users[0];
+  }
+
+  public createUser(
```

Comment:
I think so. I like the idea that knowledge about the GitLab API is not sprinkled around. We could also write an integration test for the thick GitLab-client and be sure that our GitLab API usage works.

However I don't think it's necessary to refactor the current code in one batch.